### PR TITLE
feat: add cli package

### PR DESCRIPTION
**Pull request by jaipaljadeja**

### CODE REVIEW

File: packages/cli/src/build/types.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+export async function writeTypes(apibara: Apibara) {
+  // TODO
+  generateTypes(
+    await resolveSchema(
+      Object.fromEntries(
+        Object.entries(apibara.options.runtimeConfig),
+      ) as Record<string, JSValue>,
+    ),
+    {
+      interfaceName: "NitroRuntimeConfig",
```
Comment (fracek) [inline packages/cli/src/build/types.ts:13]: Probaly `ApibaraRuntimeConfig` :)

File: packages/cli/apibara.config.ts
Comment (fracek) [inline packages/cli/apibara.config.ts:1]: I don't think we should have this `apibara.config.ts` here.

File: packages/cli/src/config/defaults.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { defaultSink } from "@apibara/indexer";
+import { defineConfig } from ".";
```
Comment (fracek) [inline packages/cli/src/config/defaults.ts:2]: Should use path to the file

File: packages/cli/src/types/config.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+export interface ApibaraConfig<
+  // biome-ignore lint/complexity/noBannedTypes: <explanation>
+  T extends Record<string, Partial<ApibaraConfig<T>>> = {},
+> extends DeepPartial<Omit<ApibaraOptions<T>, "preset" | "presets">>,
+    C12InputConfig<ApibaraConfig> {
+  sink?: {
+    default: Sink;
+    [key: string]: Sink;
+  };
+  dev?: boolean;
```
Comment (fracek) [inline packages/cli/src/types/config.ts:26]: What does `dev` do?

Comment (jaipaljadeja) [reply packages/cli/src/types/config.ts:26]: > What does dev do?

oh my bad, it should be just in `ApibaraOptions` and not in `ApibaraConfig`. 
its more like an internal flag, to determine if its a dev command or build command.

File: packages/cli/apibara.config.ts
Comment (jaipaljadeja) [reply packages/cli/apibara.config.ts:1]: > I don't think we should have this apibara.config.ts here.

correct! i have put it just for the purpose of quick testing for the time being, i will remove it once we have a proper setup ready for tests and all.

Comment (fracek) [reply packages/cli/apibara.config.ts:1]: That's fine!

File: packages/cli/src/cli/commands/prepare.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+export default defineCommand({
+  meta: {
+    name: "prepare",
+    description: "Generate types for the project",
+  },
+  args: {
+    ...commonArgs,
+  },
+  async run({ args }) {
+    consola.start("Prepairing Types");
```
Comment (fracek) [inline packages/cli/src/cli/commands/prepare.ts:17]:
```suggestion
    consola.start("Preparing Types");
```

**Review by fracek — COMMENTED** [review]
I tested the code and I get an error if I run the `apibara:dev` command in the example repo. I think we should use a more robust way to invoke the compiled module, for example by loading it with `createJiti` and invoking it that way. What do you think?

```
◐ Starting dev server                                                                                                                                            20:10:50
✔ Types generated                                                                                                                                               20:10:51
✔ Output directory cleaned                                                                                                                                      20:10:51
✔ Indexers built in 4269 ms                                                                                                                             apibara 20:10:55
✔ Dev server started                                                                                                                                            20:10:55
✔ Starting indexers                                                                                                                                             20:10:55
node:events:496
      throw er; // Unhandled 'error' event
      ^

Error: spawn jiti ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:292:12)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn jiti',
  path: 'jiti',
  spawnargs: [
    '/home/fra/Hacks/apibara/typescript-sdk/examples/cli/dist/main.mjs'
  ]
}

Node.js v20.12.2
```

**Review by fracek — COMMENTED** [review]
Got it to work by reinstalling. I still think it's worth exploring how feasible it is to use jiti as a library and not relying on it  being in the user path.

File: packages/cli/src/rollup/config.ts [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+runMain(command);
+`;
+
+  return {
+    input: {
+      main: "virtual:main.ts",
+    },
+    output: {
+      dir: join(apibara.options.outputDir || "dist"),
```
Comment (fracek) [inline packages/cli/src/rollup/config.ts:142]: IMHO the cli should output the build files to `.apibara/build`, that way a project can be cleaned up by deleting just one folder.

**Review by fracek — APPROVED** [review]

**Pull request merged**